src/mixed.js

```diff
@@ -540,6 +540,21 @@ const proto = (SchemaType.prototype = {
         ),
     };
   },
+
+  _isDefined(value) {
+    return value !== undefined;
+  },
+
+  defined(message = locale.defined) {
+    return this.nullable().test({
+      message,
+      name: 'defined',
+      exclusive: true,
+      test(value) {
+        return this.schema._isDefined(value);
```

Comment:
I don't think we need the indirection here, since it's unlikely anyone will override this. Let's just do the check here?

Reply:
Sure, gonna do it!

Also, do you think I should implement a version of it for arrays as well? I ask it based in this comment https://github.com/jquense/yup/issues/189#issuecomment-374701160

Reply:
@jquense  @talyssonoc  - Yes please for arrays - I exactly need this.

Reply:
^Wait - does this need more work for arrays? Seems to be working fine for me.